### PR TITLE
Feat/comment/update/delete

### PR DESCRIPTION
**Pull request by a-w-m**

### CODE REVIEW

File: src/models/comment.js [hunk 1]
```diff
@@ -6,7 +6,8 @@ const mongoose = require("mongoose");
 const commentSchema = new mongoose.Schema({
   user: { type: mongoose.Schema.Types.ObjectId, ref: User },
   text: { type: String, required: true },
-  date: { type: Date, default: Date.now }
+  date: { type: Date, default: Date.now },
+  isEdited : {type: Boolean, default: false}
```
Comment (rusye) [inline src/models/comment.js]: Is prettier broken?  Surprised it didn't edit this.

File: src/routes/comments.js [hunk 2]
```diff
@@ -140,11 +142,13 @@ router.patch('/:commentID', (req, res) => {
         return res.status(404).json({ Error: 'Comment not found' });
       }
 
-      if (req.user._id !== comment.user.toString()) {
+      if (req.user._id !== comment.user.toString()){
+
```
Comment (rusye) [inline src/routes/comments.js]: Curious why the line was added and why prettier didn't work...

File: client/src/components/main/chatwindow/Comment.js [hunk 3]
```diff
@@ -37,69 +61,83 @@ const Comment = props => {
...
-`;
+  return (
+    <Styled.CommentContainer>
+      <Styled.CommentAvatar id={id} />
+
+      <Styled.CommentNameDateWrapper>
+        <Styled.CommentName>{name}</Styled.CommentName>
+        <Styled.CommentTime>{formatDate(date)}</Styled.CommentTime>
+      </Styled.CommentNameDateWrapper>
+      {editComment ? (
```
Comment (rusye) [inline client/src/components/main/chatwindow/Comment.js:106]: Could use an empty line here

File: client/src/components/main/chatwindow/Comment.js [hunk 4]
```diff
@@ -37,69 +61,83 @@ const Comment = props => {
...
+        <Styled.CommentTextWrapper>
+          <Styled.CommentText>{text}</Styled.CommentText>
+          <Styled.CommentEdited isEdited={isEdited}>(edited)</Styled.CommentEdited>
+        </Styled.CommentTextWrapper>
+      )}
+
+      <Styled.CommentMenu show={user_id === localStorage.userId} onClick={handleMenu}>
+        ...
+      </Styled.CommentMenu>
+      {!isHidden && (
```
Comment (rusye) [inline client/src/components/main/chatwindow/Comment.js:121]: Could use an empty line here also

File: client/src/components/main/chatwindow/Comment.js [hunk 5]
```diff
@@ -37,69 +61,83 @@ const Comment = props => {
...
+        <Styled.CommentDropdown ref={dropdown}>
+          <Styled.MenuButton onClick={handleEditComment}>Edit Comment</Styled.MenuButton>
+          <Styled.MenuDeleteButton onClick={handleDeleteComment}>
+            Delete Comment
+          </Styled.MenuDeleteButton>
+        </Styled.CommentDropdown>
+      )}
+      {deleteComment && (
```
Comment (rusye) [inline client/src/components/main/chatwindow/Comment.js:130]: Could use an empty line here also

File: client/src/components/main/chatwindow/CreateComment.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import React, { useState, useContext } from 'react';
-import styled from 'styled-components';
+import Styled from './styles/comment.styles';
```
Comment (rusye) [inline client/src/components/main/chatwindow/CreateComment.js:2]: Hmm should we have a vote on this?  Should all styles be there own files?  I say yes and that they should live in the component folder and not a separate folder that way it'll be easier to locate.

File: client/src/components/main/chatwindow/DeleteComment.js [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+  const handleDelete = e => {
+    axios
+      .delete(`/api/comments/${id}`, {
+        headers: { authorization: `bearer ${localStorage.authToken}` },
+      })
+      .then(() => {
+        dispatch({ type: 'DELETE_FROM_DB', text });
+      })
+      .catch(err => console.error(err));
```
Comment (rusye) [inline client/src/components/main/chatwindow/DeleteComment.js:31]: I read somewhere we don't to display the error to the user on the front, we need to dumb it down.

File: client/src/components/main/chatwindow/DeleteComment.js [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+            <Styled.CommentEdited isEdited={isEdited}>(edited)</Styled.CommentEdited>
+          </CommentText>
+        </CommentContainer>
+        <ButtonWrapper>
+          <Styled.Button onClick={() => setDeleteComment(false)}>Cancel</Styled.Button>
+          <Styled.DeleteButton onClick={handleDelete}>Delete</Styled.DeleteButton>
+        </ButtonWrapper>
+      </DeletePopout>
+      {deleteComment && <Overlay></Overlay>}
+    </div>
```
Comment (rusye) [inline client/src/components/main/chatwindow/DeleteComment.js:60]: Can we get some empty lines in there to help make the code more visually appealing?

File: client/src/components/main/chatwindow/DeleteComment.js [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+  display: block; /* Hidden by default */
+  width: 100%; /* Full width (cover the whole page) */
+  height: 100%; /* Full height (cover the whole page) */
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: rgba(0, 0, 0, 0.5); /* Black background with opacity */
+  cursor: pointer; /* Add a pointer on hover */
+`;
```
Comment (rusye) [inline client/src/components/main/chatwindow/DeleteComment.js:146]: Pull out styles into its own separate file?

File: client/src/components/main/chatwindow/EditComment.js [hunk 10]
```diff
@@ -0,0 +1,74 @@
+import React, { useState, useContext } from 'react';
+import styled from 'styled-components';
+import Styled from './styles/comment.styles';
+import axios from 'axios';
+import { ChatContext } from './ChatWindow';
+
+/*
+Posts new comment to database when enter key pressed and triggers re-render by updating ChatWindow state
```
Comment (rusye) [inline client/src/components/main/chatwindow/EditComment.js]: Are we posting here?  This is a `.patch()`

File: client/src/components/main/chatwindow/EditComment.js [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+  };
+
+  const handleOnChange = event => {
+    setComment(event.target.value);
+  };
+
+  return (
+    <Styled.CommentFormWrapper className={className}>
+      <Form onSubmit={handleSubmit}>
+        <TextArea onChange={handleOnChange} onKeyDown={handleEnter} value={comment}></TextArea>
```
Comment (rusye) [inline client/src/components/main/chatwindow/EditComment.js]: You don't need `</TextArea>` in the end, you can immediately close it with `/>` at the end

File: client/src/components/main/chatwindow/CreateComment.js (hunk 6 above)
Comment (Igor333m) [reply client/src/components/main/chatwindow/CreateComment.js:2]: You are saying that comment.styles file need to be moved to chatwindow folder?

Comment (Junjpan) [reply client/src/components/main/chatwindow/CreateComment.js:2]: For now, I think we can just leave it alone to minimize the conflict. Once everyone finish their styling and are ready to go to production , we can move those styles to their own files under their own component if we decide to go that round.

File: client/src/components/main/chatwindow/DeleteComment.js (hunk 7 above)
Comment (Junjpan) [reply client/src/components/main/chatwindow/DeleteComment.js:31]: How to dumb it down if you don't display the error to the front? Any suggestion?

File: src/models/comment.js (hunk 1 above)
Comment (Junjpan) [reply src/models/comment.js]: Did you remember I mentioned to you last time, prettier made all my single quote turn into double quotes inside the form element?

Comment (a-w-m) [reply src/models/comment.js]: Does prettier edit this on your end?

File: client/src/components/main/chatwindow/CreateComment.js (hunk 6 above)
Comment (a-w-m) [reply client/src/components/main/chatwindow/CreateComment.js:2]: I made styles I reused their own file (`comment.styles`). For styles I didn't reuse, such as the style for `ChatWindow` component, I left it in the component file itself.

File: client/src/components/main/chatwindow/DeleteComment.js (hunk 8 above)
Comment (a-w-m) [reply client/src/components/main/chatwindow/DeleteComment.js:60]: Should I add an empty line between the outer level components?

File: client/src/components/main/chatwindow/EditComment.js (hunk 10 above)
Comment (a-w-m) [reply client/src/components/main/chatwindow/EditComment.js]: Yes, you're right, should be patch

File: client/src/components/main/chatwindow/DeleteComment.js (hunk 7 above)
Comment (rusye) [reply client/src/components/main/chatwindow/DeleteComment.js:31]: @Junjpan Don't `console.log` it to the user just say "Something went wrong" and log it on the backend.

File: client/src/components/main/chatwindow/DeleteComment.js (hunk 8 above)
Comment (rusye) [reply client/src/components/main/chatwindow/DeleteComment.js:60]: I'll have to see it to understand it

File: src/models/comment.js (hunk 1 above)
Comment (rusye) [reply src/models/comment.js]: @a-w-m Can I commit to your PR?

File: client/src/components/main/chatwindow/CreateComment.js (hunk 6 above)
Comment (rusye) [reply client/src/components/main/chatwindow/CreateComment.js:2]: I'll have to dig into this a little bit more, it's just starting to look messier and messier for the structure. In my opinion but we can look into this later.

File: src/models/comment.js (hunk 1 above)
Comment (rusye) [reply src/models/comment.js]: @Junjpan it didn't change something elsewhere.  I think the double quotes are because of react I think.  Let me look into this.

**Review by Junjpan — APPROVED** [review]

**Review by Igor333m — APPROVED** [review]
